platform/windows/detect.py

```diff
@@ -530,27 +513,26 @@ def configure_mingw(env):
     if env["arch"] in ["x86_32", "x86_64"]:
         env["x86_libtheora_opt_gcc"] = True
 
-    mingw_bin_prefix = get_mingw_bin_prefix(env["mingw_prefix"], env["arch"])
-
     if env["use_llvm"]:
-        env["CC"] = mingw_bin_prefix + "clang"
-        env["CXX"] = mingw_bin_prefix + "clang++"
-        if try_cmd("as --version", env["mingw_prefix"], env["arch"]):
-            env["AS"] = mingw_bin_prefix + "as"
-        if try_cmd("ar --version", env["mingw_prefix"], env["arch"]):
-            env["AR"] = mingw_bin_prefix + "ar"
-        if try_cmd("ranlib --version", env["mingw_prefix"], env["arch"]):
-            env["RANLIB"] = mingw_bin_prefix + "ranlib"
+        env["CC"] = get_mingw_tool("clang", env["mingw_prefix"], env["arch"])
+        env["CXX"] = get_mingw_tool("clang++", env["mingw_prefix"], env["arch"])
+        tool_as = get_mingw_tool("as", env["mingw_prefix"], env["arch"])
+        tool_ar = get_mingw_tool("ar", env["mingw_prefix"], env["arch"])
+        tool_ranlib = get_mingw_tool("ranlib", env["mingw_prefix"], env["arch"])
```

Comment:
I suppose not, but could it have a measurable performance impact that we're now doing the full logic of `get_mingw_tool` 5 times, instead of just `get_mingw_bin_prefix` once to reuse?

Reply:
It's just calling `--version` on the tools, so I strongly doubt it'd be an issue